### ISSUE TITLE
Audible and spoken cues for blind subtitle editors

### ISSUE THREAD

**Issue by Piciok** [SubtitleEdit/subtitleedit#2661]
In the current state it is possible for a blind person to use the Subtitle Edit software with a screenreader such as NVDA. As it is built based on standard Windows controls all the windows and lists are accessible and the presence of configurable hotkeys allows the blind to perform desired actions much faster.

The only issue I've run into is not being able to tell whether my added subs display at the right time in the sort of WYSIWYG way.

I have thought about it a little and here is my proposed solution:

- the moment of the subtitle appearing on the screen as well as disappearing from it could be indicated with a short sound so that the time frames of the subtitle's presence is easy to grasp for those who can't see it on-screen;
- as soon as the sub appears, it should be read out by the screenreader used by the blind person. This can be achieved with the help of the Tolk library, to be found under
https://github.com/dkager/tolk
As Tolk can handle all the popular screenreaders +SAPI output, shall no other be present, it's just a matter of having Subtitle Edit send the subtitle to Tolk.
- an additional soundcue could be added as the subtitle is read out to indicate the place where it wraps to a new line (a page-flipping or similar sound) so that the blind person could have the idea how much of the subtitle is visible to the viewer at once and decide about the text proportions;

By implementing the suggested improvements it would be possible to enable blind translators to independently create subtitles for movies which opens new opportunities both in the designated  workshops and classes as well as on the job market where they can provide translations for movies in the form of subtitles.

**Comment by darnn** [comment]
I don't know how much of a priority this is for Nik, but if you ever want to pay someone to write and submit the code that could do this, I'd love to chip in.

**Comment by Piciok** [comment]
As far as the speaking part, a workaround that is easier to implement and still does the trick is to create an option of copying the currently displayed subtitle to the title of the Subtitle Edit window. Such changes are reported by screenreaders as they happen both in speech and braille and the implementation doesn't require any external dependencies.

**Comment by Piciok** [comment]
Hello again,

I have opened a discussion on possible integration of the proposed changes in the core of the free and opensource NVDA screenreader in the form of an integrated NVDA app module.

The discussion is here
https://github.com/nvaccess/nvda/issues/7871

It was suggested that further discussion be moved to this issue tracker. Also, if you cannot contribute at this time, you can contact NVAccess for further assistance.
